src/kemalyst-model.cr

```diff
@@ -101,99 +111,104 @@ class Kemalyst::Model
       {% end %}
       return params
    end
-  end # End of Fields Macro
 
-  # Clear is used to remove all rows from the table and reset the counter for
-  # the id.
-  def self.clear
-    @@adapter.clear @@table_name
-  end
+    # Clear is used to remove all rows from the table and reset the counter for
+    # the primary key.
+    def self.clear
+      @@adapter.clear @@table_name
+    end
 
-  # The save method will check to see if the @id exists yet.  If it does it
-  # will call the update method, otherwise it will call the create method.
-  # This will update the timestamps apropriately.
-  def save
-    begin
-      if value = @id
-        @updated_at = Time.now
-        params_and_id = params
-        params_and_id << value
-        @@adapter.update @@table_name, self.class.fields, params_and_id
-      else
-        @created_at = Time.now
-        @updated_at = Time.now
-        @id = @@adapter.insert @@table_name, self.class.fields, params
-      end
-      return true
-    rescue ex
-      if message = ex.message
-        puts "Save Exception: #{message}"
-        errors << Kemalyst::Validators::Error.new(:base, message)
+    # The save method will check to see if the primary exists yet. If it does it
+    # will call the update method, otherwise it will call the create method.
+    # This will update the timestamps apropriately.
+    def save
+      begin
+        if value = @{{primary_name}}
+          @updated_at = Time.now
+          params_and_pk = params
+          params_and_pk << value
+          @@adapter.update @@table_name, @@primary_name, self.class.fields, params_and_pk
+        else
+          @created_at = Time.now
+          @updated_at = Time.now
+          {% if primary_type.id == "Int32" %}
+            @{{primary_name}} = @@adapter.insert(@@table_name, self.class.fields, params).to_i32
+          {% else %}
+            @{{primary_name}} = @@adapter.insert(@@table_name, self.class.fields, params)
+          {% end %}
```

Comment:
I could see someone wanting to use a UUID as a primary key but not sure how to implement that.  Maybe something for a future change.